classes/salesforce_mapping.php

```diff
@@ -57,14 +58,16 @@ class Object_Sync_Sf_Mapping {
 	 * @param object $wpdb A WPDB object.
 	 * @param string $version The plugin version.
 	 * @param string $slug The plugin slug.
+	 * @param string $option_prefix The plugin option prefix
 	 * @param object $logging Object_Sync_Sf_Logging.
 	 * @throws \Exception
 	 */
-	public function __construct( $wpdb, $version, $slug, $logging ) {
-		$this->wpdb    = $wpdb;
-		$this->version = $version;
-		$this->slug    = $slug;
-		$this->logging = $logging;
+	public function __construct( $wpdb, $version, $slug, $option_prefix, $logging ) {
```

Comment:
Since `__construct` is a public method and this is the 1.4 release (not a 2.0 backwards compatibility breaking release), the new parameters should be optional. We can't guarantee developers haven't invoked this method.

Here's one example I can say for sure has been utilized to extend the plugin. We process woocommerce orders using a custom SF REST method, which handles complex business rules and creates a variety of SF Objects. We then connect the new SF Objects to their corresponding WP records to save time searching during future syncs. Since there was no high level function for that, we construct a Object_Sync_Sf_Mapping to access the create_object_map method.

Reply:
This is great to know. I'm assuming that if I put `$option_prefix = ''` at the end, this will continue to work for you. As in bb37b25543767282656da53140f58991e953f581.

Reply:
Making the new arguments optional and at the end certainly works for me! Do they need to run through the same setting filter as the base class to initialize the values, if they arrive as the default?

Reply:
Sorry, can you say more about what you mean by that?

Reply:
In `object-sync-for-salesforce.php` line 135 (https://github.com/MinnPost/object-sync-for-salesforce/pull/195/files#diff-1e6eb137ec04d15b065e6fc5822ca51cR135), the option_prefix is set to `object_sync_for_salesforce_`, which ideally passes through to influence the options, filters, and action across the rest of the plugin.

I was trying to say that value might be a better argument default than the empty string. The empty string could lead to classes using incorrect option, hook, and action strings.

Reply:
That does make sense. I put that same default in each class where it could come in as an empty value. ed7f59ca2f77e69b03c18fca0cf686005a5b21c8